code/game/machinery/ftl_shieldgen.dm

```diff
@@ -89,6 +99,8 @@
 	if(stat & (BROKEN|MAINT))
 		charging_plasma = 0
 		return
+	if(plasma_charge > 0)
+		plasma_charge -= min(plasma_passiveloss, plasma_charge)
```

Comment:
I dislike the complete disappearance of plasma even if the shield is off. You also have code here meaning if the shield is broken as opposed to just off it doesn't lose any plasma.